### DISCUSSION TITLE
Choosing model switching strategy

### DISCUSSION THREAD

**Discussion by micbac159** [asreview/asreview#1423] [Q&A]
I've reached my stopping criterion, and I would like to switch to another model. 

I've used the standard settings (Naïve Bayes, TF-IDF, Max) so far. However, I struggle a bit with defining which model I should go for. I thought of trying no more than two models for another 10% of the sample or unless I do not get any relevant results. 

The next natural step seems to be going for a combination of Doc2Vec + CNN. But I cannot really justify why this would be the best choice.

I did some search on the forum and found this paper: 'Active learning-based Systematic reviewing using switching classification models: the case of the onset, maintenance, and relapse of depressive disorders' (Link: https://psyarxiv.com/t7bpd). However, I could not really judge based on the simulation outcome, what model would be suitable to go for in the second iteration because I'm reviewing social science papers, which have not been thoroughly tested in the above mentioned simulation study.

Does anyone have ideas/experience with model switching strategies? 

Also, I'm planning to use my labelled data from the initial set-up to feed the next model. Any pros and cons of that approach?

**Comment by rohitgarud** [comment]
Hi @micbac159,
I would suggest running a few simulations with the already labeled records from your dataset with different settings and checking whether the performance for different combinations of features and classifiers correlates well with the results in the paper you mentioned. The simulations should not take much time and can give you some idea about the performance of different settings for your specific domain.

**Reply by Rensvandeschoot** [reply]
I would suggest considering a different feature extractor to improve your model's performance. Instead of using the default TF-IDF feature extraction method, you can try alternatives like sBert or Doc2Vec. 

sBert (Sentence-BERT) is a modification of the BERT model that enables you to obtain fixed-sized sentence embeddings, which can be beneficial for capturing semantic information in the text. On the other hand, Doc2Vec is an unsupervised algorithm that generates document embeddings and can capture the context and meaning of the text more effectively than the bag-of-words approaches like TF-IDF.

Considering a different feature extractor like sBert or Doc2Vec leads to a re-computed feature matrix, which may result in a different rank ordering of the papers you have not yet seen as compared to continuing with the old model. This change in rank ordering could provide new insights into the relevance of the unseen papers, potentially leading to more efficient labeling and better model performance.
